src/wp-includes/class-wp-widget.php

```diff
@@ -612,12 +612,16 @@ public function get_settings() {
 		$settings = get_option( $this->option_name );
 
 		if ( false === $settings ) {
+			$settings = array();
 			if ( isset( $this->alt_option_name ) ) {
-				$settings = get_option( $this->alt_option_name );
-			} else {
-				// Save an option so it can be autoloaded next time.
-				$this->save_settings( array() );
+				// Get settings from legacy (alternative) option.
```

Comment:
Nit:
- This reads "legacy (alternative) option"
- Line 620 reads "legacy option"
- The tests read "alt (legacy) option".